### DISCUSSION TITLE
Selecting device in VectorUnitCellUpdatePolicy

### DISCUSSION THREAD

**Discussion by KunalRandad** [IBM/aihwkit#643] [General]
I want to simulate a cell by 3 devices. So VectorUnitCellUpdatePolicy looked like a good solution. The devices are supposed to have scaling factors of 100, 10  and 1 (say) and then, the scaled output will be summed. However, for each update in backward pass, I want to have control over which device I want to update. I want to update the cell by selecting device as per the gradient (only sign without magnitude also works for me) and the current device state.  If possible, I want to have control on how each increment or decrement pulse may be applied to the device.  The problem I am facing is that there does not seem to be any way to control device, even when using SINGLE_FIXED. I could not find any variable or function to control the device being updated. SingleFixed method just uses a constant index. I would like to know how can I implement my requirements.

**Comment by maljoras-sony** [comment]
Hi @KunalRandad,
many thanks for the question. Actually, there is a way to control which device is updated. You can use `analog_tile.set_hidden_update_index(idx)`, see here for the [implementation](https://github.com/IBM/aihwkit/blob/69e524fbc2c5462cbe1a12ee1662879c001c30c0/src/aihwkit/simulator/tiles/rpucuda.py#L230) and the [documentation](https://aihwkit.readthedocs.io/en/latest/api/aihwkit.simulator.tiles.rpucuda.html#aihwkit.simulator.tiles.rpucuda.RPUCudaSimulatorTileWrapper.set_hidden_update_index)
